src/events/upgrade.py

```diff
@@ -120,15 +120,20 @@ def current_version(self) -> str:
         dependency_model: DependencyModel = getattr(self.dependency_model, "kafka_service")
         return dependency_model.version
 
+    @property
+    def zookeeper_current_version(self) -> str:
+        """Get current Zookeeper version."""
+        return self.charm.state.zookeeper.zookeeper_version
+
```

Comment:
todo: I think this is redundant, and splits where we reference state. I think we should remove it and just use `self.charm.state.zookeeper.zookeeper_version` as normal.

Reply:
This was taken from vm

Reply:
Looking at where it's used it can probably be simplified and just mock the state directly on the test.

Reply:
That's fine. I don't mind, but please update VM as well to keep the two consistent.